earth_enterprise/src/common/CacheSizeCalculations.h

```diff
@@ -19,29 +19,77 @@
 #include "khTypes.h"
 #include <memory>
 #include <vector>
+#include <set>
+#include <map>
+#include <string>
+#include "notify.h"
+#include <qstring.h>
+#include <typeinfo>
+#include <type_traits>
+#include "SharedString.h"
 
-// determine amount of memory used by generic objects
+//determine amount of memory used by an object pointed to by a pointer
 template<class T>
-inline uint64 GetObjectSize(const T &obj) {
-    return sizeof(obj);
+inline uint64 GetHeapUsage(const T* &ptr) {
+    uint64 ptrSize = (ptr ? GetSize(*ptr) : 0);
+    return ptrSize;
 }
-// determine amount of memory used by a vector and all of it's contents
+//determine amount of memory used by an object pointed to by a shared_ptr
 template<class T>
-inline uint64 GetObjectSize(const std::vector<T> &vec) {
+inline uint64 GetHeapUsage(const std::shared_ptr<T> &shptr) {
+    uint64 shptrSize = (shptr ? GetSize(*shptr) : 0);
+    return shptrSize;
+}
+//determine amount of memory used by a string
+inline uint64 GetHeapUsage(const std::string &str) {
+    uint64 strSize = (str.capacity() * sizeof(char));
+    return strSize;
+}
+// determine amount of memory used by a qstring
+inline uint64 GetHeapUsage(const QString &qstr) {
+    uint64 qstrSize = (qstr.capacity() * sizeof(char16_t));
+    return qstrSize;
+}
+// determine amount of memory used by a given object
+// returns 0 for all non specified objects
+template<class T>
+inline uint64 GetHeapUsage(const T &obj) {
+    notify(NFY_WARN, "\tGENERAL of %s", typeid(obj).name());
```

Comment:
Minor: I think we'll want to remove this before checking in the code. This logging statement would show up a lot, and in many cases nothing would be wrong.